server/src/test/java/org/elasticsearch/index/shard/IndexShardIT.java

```diff
@@ -771,14 +770,14 @@ public void testPendingRefreshWithIntervalChange() throws Exception {
         // wait for both to ensure we don't have in-flight operations
         updateSettingsLatch.await();
         refreshLatch.await();
-        // ensure no scheduled refresh to compete with the scheduleRefresh we are going to verify.
-        assertBusy(() -> {
-            for (ThreadPoolStats.Stats stat : indexService.getThreadPool().stats()) {
-                if (stat.getName().equals(ThreadPool.Names.REFRESH) && (stat.getQueue() > 0 || stat.getActive() > 0)) {
-                    throw new AssertionError(); // cause assert busy to retry
-                }
-            }
-        });
+        // We need to ensure a `scheduledRefresh` triggered by the internal refresh setting update is executed before we index a new doc;
+        // otherwise, it will compete with the `scheduledRefresh` that we are going to verify. Since the refresh in single-node tests
+        // is a single thread executor, we can make sure that all scheduled refresh tasks are done by submitting a task then wait until
```

Comment:
why is it a single thread executor? I can't see where this is enforced

Reply:
Sorry, I misread this.

Reply:
fixed in https://github.com/elastic/elasticsearch/pull/45025/commits/6c93e23c0019fa5b818d178a9528fb4d43dc7f81